Paco-Server/src/com/google/sampling/experiential/server/CloudSqlSearchServlet.java

```diff
@@ -185,18 +201,25 @@ private String getStackTraceAsString(Throwable e) {
     return string;
   }
   
-  private boolean isTimezoneNeeded(String[] projCols){
-    boolean timezoneNeeded = false;
-    for (String eachCol : projCols) {
-      if (eachCol.equals(Constants.STAR)){
-        return false;
-      } else if(dateColumns.contains(eachCol)){
-        return true;
-      } 
+  private boolean isValidGroupBy(String[] selectColumnsArr, List<String> groupByCols) {
+    // all (plain) columns in projection (except aggregate functions on columns) must be in group by list
+    // select experiment_id, count(experiment_version) from events group by who INVALID
+    // plain columns->experiment_id; aggregate columns->experiment_version); group by column->who
+    // select who, count(experiment_version) from events group by who VALID
+    boolean isValidGroupBy = true;
+    List<String> selectColumns = Lists.newArrayList();
+    for(String s: selectColumnsArr) {
```

Comment:
If I do it in jsql parser, then I have to capture what column names are part of sub query, main query, inner query etc. I need to do this validation for just the main query. Since I have already captured it in the initial object, I have used it instead of doing additional work in jsql parser.